internal/worker_runner.go

```diff
@@ -313,10 +314,21 @@ func (wr *workerRunner) queryOBD() {
 				continue
 			}
 		}
+		// check if we have failed to query this pid too many times
+		if wr.signalsQueue.failureCount[request.Name] > 10 {
+			continue
+		}
+
 		// execute the pid
 		obdResp, ts, err := commands.RequestPIDRaw(&wr.logger, wr.device.UnitID, request)
 		if err != nil {
 			wr.logger.Err(err).Msg("failed to query obd pid")
+			wr.signalsQueue.IncrementFailureCount(request.Name)
+			// if we failed too many times, we should send an error to the cloud
+			if wr.signalsQueue.failureCount[request.Name] > 10 {
```

Comment:
would this logging code ever get hit because it was already controlled for on line 318?

Reply:
with latest commit we improved the logging to cloud, thanks for help